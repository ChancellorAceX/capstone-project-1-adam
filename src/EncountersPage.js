import React from 'react';
import './EncounterPage.css';

export default function EncountersPage(){
  return(
    <section className='container'>
      <header className='detailheader'>
        <button className='back'>back button</button>
        <h2>Encounters</h2>
        <button className='encounter'>add button</button>
      </header>
      <body>
        <section className='campaignheader'>
          <h4 className='name'>Campaign A</h4>
          <p className='expanded'>-</p>
        </section>
        <ul className='name'>
          <li className='location'>
            Location 1
            <ul>
              <li>Sub-location 1</li>
              <li>Sub-location 2</li>
              <li>Sub-location 3</li>
            </ul>
          </li>
          <li className='location'>
            Location 2
            <ul>
                <li>
                  <details>
                    <summary>Sub-location 4</summary>
                    <details>
                      <summary>Encounter A</summary>
                      <p>Encounter A description</p>
                      <button type='button'>Detail view</button>
                    </details>
                    <p>Encounter B</p>
                  </details>
                </li>
                <li>
                <details>
                    <summary>Sub-location 5</summary>
                    <p>Encounter C</p>
                  </details>
                </li>
                <li>
                <details>
                    <summary>Sub-location 6</summary>
                    <p>Encounter D</p>
                    <p>Encounter E</p>
                  </details>
                </li>
            </ul>
          </li>
          <li className='location'>Location 3</li>
        </ul>
        <section className='campaignheader'>
          <h4 className='name'>Campaign B</h4>
          <p className='collapsed'>+</p>
        </section>
        <section className='campaignheader'>
          <h4 className='name'>Campaign C</h4>
          <p className='expanded'>+</p>
        </section>
      </body>
      <footer>

      </footer>
    </section>
  )
}