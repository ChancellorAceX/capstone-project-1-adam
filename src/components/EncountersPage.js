import React from 'react';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';
// import TokenService from '../services/token-service';
import '../styles/EncounterPage.css';
import Header from './Header';
import Campaign from './Campaign';

export default withRouter(class EncountersPage extends React.Component {
  componentDidMount() {
    this.props.fetchEncounters()
  };

  render() {
    return (
      <section className='container'>
        <Header
          pageName='Encounters'
          history={this.props.history}
        />
        <main className='encounters'>
        <section className='addButtonContainer'>
            <Link to='/new/campaign'><button type='button'>+ Campaign</button></Link>
            <Link to='/new/encounter'><button type='button'>+ Encounter</button></Link>
          </section>
          {this.props.campaigns.map(campaign =>
            <Campaign
              key={campaign[0]}
              campaigns={this.props.campaigns}
              campaign={campaign}
              encountersSource={this.props.encounters}
              encounters={this.props.encounters.filter(encounter => campaign.includes(encounter.title))}
              setTargetState={this.props.setTargetState}
            />)}
        </main>
      </section>
    );
  }
});