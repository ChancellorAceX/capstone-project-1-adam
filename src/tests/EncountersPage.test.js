import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EncountersPage from '../components/EncountersPage';

describe('EncountersPage component', () => {
  const props = {
    "campaigns": [
      "[1, \"Azeroth\"]",
      "[6, \"Fillory\"]",
      "[7, \"Grishaverse\"]"
    ],
    "encounters": [
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 6, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 7, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}",
      "{campaignuserid: 1, cid: 1, created: \"2020-07-03T00…}"
    ],
    "fetchEncounters": "ƒ () {}",
    "setEncounters": "ƒ () {}",
    "setCampaigns": "ƒ () {}",
    "setTargetState": "ƒ () {}",
    "loggedInToggle": "ƒ () {}"
  };

  it('renders the full encounters page component', () => {
    const wrapper = shallow(<EncountersPage {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});