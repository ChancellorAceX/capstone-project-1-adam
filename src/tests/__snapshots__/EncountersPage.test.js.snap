// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EncountersPage component renders the full encounters page component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
